extras/tests/unitTest/unitTest.ino

```diff
@@ -310,6 +310,65 @@ test(timer_cancel_all) {
     assertEqual(task.runs, 0UL);
 }
 
+///////////////////////////////////////////////////////////////////////////////
+// support for in-task cancel check
+bool tickA(void *);
+bool tickB(void *);
+
+unsigned long ticksA = 0;
+unsigned long ticksB = 0;
+Timer<0x5, CLOCK::millis, void *> *myTimer = 0;
+Timer<0x5, CLOCK::millis, void *>::Task pendingTask = 0;
+
+bool tickA(void *) {
+    ticksA++;
+    if (myTimer) {
+        myTimer->cancel(pendingTask);
+        pendingTask = myTimer->in(1, tickB);  // A then B
+    }
+    return false;
+}
+bool tickB(void *) {
+    ticksB++;
+    if (myTimer) {
+        myTimer->cancel(pendingTask);
+        pendingTask = myTimer->in(1, tickA);  // B then A
+    }
+    return false;
+}
+
+test(timer_cancel_in_task) {
```

Comment:
This regression test will only PASS with the change (legacy code will fail).  It shows `timer.cancel(myRunningTask)` can work and added tasks will run too.